lib/mayaUsd/render/pxrUsdMayaGL/hdImagingShapeUI.h

```diff
@@ -18,24 +18,23 @@
 
 /// \file pxrUsdMayaGL/hdImagingShapeUI.h
 
-#include "pxr/pxr.h"
-#include "../../base/api.h"
-
-// XXX: On Linux, some Maya headers (notably M3dView.h) end up indirectly
-//      including X11/Xlib.h, which #define's "Bool" as int. This can cause
-//      compilation issues if sdf/types.h is included afterwards, so to fix
-//      this, we ensure that it gets included first.
-#include "pxr/usd/sdf/types.h"
+#include <mayaUsd/base/api.h>
 
 #include <maya/M3dView.h>
 #include <maya/MDrawInfo.h>
 #include <maya/MDrawRequest.h>
 #include <maya/MDrawRequestQueue.h>
 #include <maya/MPxSurfaceShapeUI.h>
 
+#include <pxr/pxr.h>
 
-PXR_NAMESPACE_OPEN_SCOPE
+// XXX: On Linux, some Maya headers (notably M3dView.h) end up indirectly
```

Comment:
Hey @ppt-adsk , great catch. I am curious about this too. Do we know if this is really the case or at least if it has been fixed in recent versions of API?  I can put it back before M3dView.h if you guys think it is necessary.  

@mattyjams  Do you know the history about this?

Reply:
I ran across this as well at some point - and since the problem is really more with a system header, `X11/Xlib.h`, I doubt this has gone away (at least not for us, since we're still using the same Linux distribution / version!) - it would only no longer be an issue if `M3dView` stopped including `X11/Xlib.h`.  Even if it has, we'd have to put in a conditional include, at which point it seems simpler to just force the ordering.

Unfortunately, this would deviate from the "standard" ordering.  Does anyone know how clang-format handles `#include`s with comments above them? I'm fairly certain it's intelligent about not reordering #includes inside of conditional blocks... but don't remember how it handles `#include` comments like this.  (Worst case, we'd have to protect with `// clang-format off`.)

Alternatively - it's not too late to swap the order of pxr and maya - the ordering of pxr vs maya headers was pretty arbitrary, and if we have a known case where pxr needs to come first, that seems as good a reason as any to say pxr then maya.  We'd have to alter doc/codingGuidelines.md, but that hasn't been released for long (and never on master branch), so I don't see that as a roadblock.

Reply:
Hmm, good to know that this shouldn't be a problem >= 2019!  Maybe we can move it after, and add a note so we can remove the speical handling once we drop 2018 support?